### DISCUSSION TITLE
Incorrect start of week in journal pages

### DISCUSSION THREAD

**Discussion by rolfkleef** [zim-desktop-wiki/zim-desktop-wiki#1954] [Q&A] [answered]
I'm trying to get the start of the week set to Monday in the weekly pages in the Journal plugin.

From the discussion in #632 I get the impression that in principle it should pick up the locale setting.

In my case, the calendar widget seems to do this alright, but the date range in the journal template page still starts on Sunday:
![Screenshot from 2022-02-24 13-29-55](https://user-images.githubusercontent.com/223935/155524339-0460600d-e0f8-4e49-911a-f535b0bad1ca.png)

My locale is all set to en_IE.UTF-8, running Zim 0.74.3 on Ubuntu 21.10.

I've been so far using the %U format for week numbers as a workaround, but would now like to figure out why the weeks still start on Sunday. I've looked a bit at the code and tried to run with the debug flag, but somehow don't see anything that sheds more light on it.

**Comment by introt** [comment]
Please do try out #1732, as its author has not. If it fixes the problem, please open a new PR against `develop`.

**Reply by rolfkleef** [reply]
Thanks, that seems to fix it indeed. New pull request #1961 against `develop`.

**Reply by rolfkleef** [reply]
I've actually found the real cause and solution in my case, and rewrote my PR to be a very different suggestion...

**Comment by rolfkleef** [comment, marked as answer]
To document here: I found out that I did not have the Babel python library properly installed. Running `pip3 install Babel` (Ubuntu 21.10) turned out to be the real fix.

**Comment by jefftemplon** [comment]
Anyone know how this would work if one uses the MacOS app?  It appears to have a self-contained python, and no pip.

**Reply by dehesselle** [reply]
I can integrate that into the app. Babel isn't listed as a (optional) dependency anywhere, so I wasn't aware.
